File: Books/Math_Refresher/python/normal_distribution.py
# normal_distribution.py
# Normal Distribution
#-------------------------------------------------------------------------
import numpy as np
import matplotlib.pyplot as plt
from scipy.stats import norm

plt.close('all')

# Set seed for reproducibility (similar to rng default in MATLAB)
np.random.seed(0)

# Generate two sets of random numbers from normal distributions:
# r1 ~ N(0,1) and r2 ~ N(0,2)
r1 = np.random.normal(0, 1, 1000)
r2 = np.random.normal(0, 2, 1000)

# Generate fitted curves using the normal pdf.
# Here we use a common x-range for both curves.
x_range = np.linspace(-6, 6, 200)
y1 = norm.pdf(x_range, loc=0, scale=1)
y2 = norm.pdf(x_range, loc=0, scale=2)

plt.figure()
plt.plot(x_range, y1, ':', linewidth=2, label=r'$$\mu=0,\sigma^2=1$$')
plt.plot(x_range, y2, '-.', linewidth=2, label=r'$$\mu=0,\sigma^2=2$$')
plt.xlabel(r'$$x$$', fontsize=12)
plt.ylabel(r'$$f(x)$$', fontsize=12)
plt.title(r'$$f(x)$$', fontsize=12)
plt.legend(loc='upper right', fontsize=10)
plt.xlim([-6, 6])
plt.savefig("../figures/normal_distribution.pdf")
plt.show()

File: Books/Math_Refresher/matlab/normal_distribution.m
% normal distribution


% Start of script
%-------------------------------------------------------------------------%
close all;                   	% close all figures
clearvars; clearvars -global;	% clear all variables
clc;                         	% clear the command terminal
format shortG;                 	% pick the most compact numeric display
format compact;                	% suppress excess blank lines

%f(x) = randn(x)
rng default; % For reproducibility
r1 = normrnd(0,1,1000,1);
r2 = normrnd(0,2,1000,1);
h = histfit(r1);
x1 = h(2, 1).XData;
y1 = h(2, 1).YData;
h = histfit(r2);
x2 = h(2, 1).XData;
y2 = h(2, 1).YData;
close;

figure(1)
plot(x1,y1,':',x2,y2,'-.','LineWidth',2)
xlabel('$$x$$','Interpreter','latex')
ylabel('$$f(x)$$','Interpreter','latex')
title('$$f(x)$$','Interpreter','latex')
legend('$$\mu=0,\sigma^2=1$$','$$\mu=0,\sigma^2=2$$','Interpreter','latex')
xlim([-6 6])

save_all_figs_OPTION('../figures/normal_distribution','pdf')